tests/lib/rules/define-props-declaration.js

```diff
@@ -162,6 +633,26 @@ tester.run('define-props-declaration', rule, {
           line: 3
         }
       ]
+    },
+    // array
+    {
+      filename: 'test.vue',
+      code: `
+      <script setup lang="ts">
+      const props = defineProps([kind])
```

Comment:
The array elements must be strings. The `kind` variable shouldn't auto-fix because we don't know what's actually in it.
Could you leave this test as a test case for when not to auto-fix and add a new test case using a string literal?

https://vuejs.org/guide/components/props.html#props-declaration

Reply:
My bad, you're right. It does not default to a `string`. Leaving it unhandled then.

Reply:
Could you please add the test case you requested earlier?

```vue
      <script setup lang="ts">
      const props = defineProps(['kind'])
      </script>
```

https://github.com/vuejs/eslint-plugin-vue/pull/2466#discussion_r1676641695